terra-workspace-manager-jmeter/src/main/java/automation/jmeter/CreateWorkspaceSimulation.java

```diff
@@ -0,0 +1,307 @@
+        setupThreadGroup.setIsSameUserOnNextIteration(true);
+        setupThreadGroup.setProperty(SetupThreadGroup.ON_SAMPLE_ERROR, SetupThreadGroup.ON_SAMPLE_ERROR_CONTINUE);
+        setupThreadGroup.setSamplerController(setUpLoopController);
+        setupThreadGroup.setProperty(TestElement.TEST_CLASS, SetupThreadGroup.class.getName());
+        setupThreadGroup.setProperty(TestElement.GUI_CLASS, SetupThreadGroupGui.class.getName());
+    }
+
+    private static void setUpLoopController() {
+        setUpLoopController = new LoopController();
+        setUpLoopController.setName("setUp Loop controller");
+        setUpLoopController.setLoops(1);
+        setUpLoopController.setContinueForever(false);
+        setUpLoopController.setFirst(true);
+        setUpLoopController.setProperty(TestElement.TEST_CLASS, LoopController.class.getName());
+        setUpLoopController.setProperty(TestElement.GUI_CLASS, LoopControlPanel.class.getName());
+        setUpLoopController.initialize();
+    }
+
+    private static void setJSR223PreProcessor() {
+        jsr223PreProcessor = new JSR223PreProcessor();
+        jsr223PreProcessor.setName("JSR223 Pre Processor");
+        jsr223PreProcessor.setEnabled(true);
+        jsr223PreProcessor.setProperty("cacheKey", "true");
+        jsr223PreProcessor.setProperty("script", "${__setProperty(oauth2Token, ${__OAuth2Token(GCP, FIRECLOUD_SERVICE_ACCOUNT_CREDS, SAM_USER)})}");
+        jsr223PreProcessor.setProperty("scriptLanguage", "groovy");
+        jsr223PreProcessor.setProperty(TestElement.TEST_CLASS, JSR223PreProcessor.class.getName());
+        jsr223PreProcessor.setProperty(TestElement.GUI_CLASS, TestBeanGUI.class.getName());
+    }
+
+    private static void setThreadGroup(int noOfThreads, int setRamupNo) {
+        threadGroup = new ThreadGroup();
+        threadGroup.setName("CreateWorkspace Thread Group");
+        threadGroup.setNumThreads(noOfThreads);
+        threadGroup.setRampUp(setRamupNo);
+        threadGroup.setScheduler(false);
+        threadGroup.setIsSameUserOnNextIteration(false);
+        threadGroup.setProperty(ThreadGroup.ON_SAMPLE_ERROR, ThreadGroup.ON_SAMPLE_ERROR_CONTINUE);
+        threadGroup.setSamplerController(loopController);
+        threadGroup.setProperty(TestElement.TEST_CLASS, ThreadGroup.class.getName());
+        threadGroup.setProperty(TestElement.GUI_CLASS, ThreadGroupGui.class.getName());
+    }
+
+    private static void setHttpSampler(String requestType, String setPath) {
+        httpSamplerProxy = new HTTPSamplerProxy();
+        httpSamplerProxy.setDomain("${__env(WSM_SERVER)}");
+        httpSamplerProxy.setProperty("HTTPSampler.port", "${__env(WSM_PORT)}");
+        httpSamplerProxy.setProperty("HTTPSampler.protocol", "${__env(WSM_PROTOCOL)}");
+        httpSamplerProxy.setPath(setPath);
+        httpSamplerProxy.setMethod(requestType);
+        String body = "{" + "\"id\"" + ":" + " \"" + "${workspaceId}" + "\", " + "\"authToken\"" + ":" + " \""
```

Comment:
We have a model for the `CreateWorkspaceRequestBody` that can be (mapped into a string and) reused here.

https://github.com/DataBiosphere/terra-workspace-manager/blob/dev/src/main/resources/api/service_openapi.yaml#L390

Reply:
Is everyone using `io.swagger.codegen` for this? Is there an example from Unit or Integration tests I can reuse, otherwise I can figure this out.

Reply:
Here's an example https://github.com/DataBiosphere/terra-workspace-manager/blob/dev/src/test/java/bio/terra/workspace/service/workspace/WorkspaceServiceTest.java#L122-L127

I don't know that all teams are currently using Swagger codegen, but I believe the future plan is to align on it. @gjuggler may be able to speak more to that.